pkg/apis/extensions/v1beta1/defaults.go

```diff
@@ -77,16 +77,20 @@ func SetDefaults_Deployment(obj *Deployment) {
 			strategy.RollingUpdate = &rollingUpdate
 		}
 		if strategy.RollingUpdate.MaxUnavailable == nil {
-			// Set default MaxUnavailable as 1 by default.
-			maxUnavailable := intstr.FromInt(1)
+			// Set default MaxUnavailable as 25% by default.
+			maxUnavailable := intstr.FromString("25%")
 			strategy.RollingUpdate.MaxUnavailable = &maxUnavailable
 		}
 		if strategy.RollingUpdate.MaxSurge == nil {
-			// Set default MaxSurge as 1 by default.
-			maxSurge := intstr.FromInt(1)
+			// Set default MaxSurge as 25% by default.
+			maxSurge := intstr.FromString("25%")
 			strategy.RollingUpdate.MaxSurge = &maxSurge
 		}
 	}
+	if obj.Spec.RevisionHistoryLimit == nil {
+		obj.Spec.RevisionHistoryLimit = new(int32)
+		*obj.Spec.RevisionHistoryLimit = 3
```

Comment:
is this int a pointer only because we can detect if the user didn't specify a value and put a default ?